pandas/io/_plugin_loader.py

```diff
@@ -92,49 +95,82 @@ def _create_series_writer_function(format_name):
     When calling `Series.to_<whatever>` we call the dataframe writer, so
     we need to convert the Series to a one column dataframe.
     """
+
     def series_writer_wrapper(self, *args, **kwargs):
         dataframe_writer = getattr(self.to_frame(), f"to_{format_name}")
         dataframe_writer(*args, **kwargs)
 
     return series_writer_wrapper
 
 
+def _warn_conflict(func_name, format_name, loaded_plugins, module):
+    package_to_load = importlib_metadata.packages_distributions()[module.__name__]
+    if format_name in loaded_plugins:
+        # conflict with a third-party connector
+        loaded_module = loaded_plugins[format_name]
+        loaded_package = importlib_metadata.packages_distributions()[
+            loaded_module.__name__
+        ]
+        msg = (
+            f"Unable to create `{func_name}`. "
+            f"A conflict exists, because the packages `{loaded_package}` and "
+            f"`{package_to_load}` both provide the connector for the '{format_name}' format. "
+            "Please uninstall one of the packages and leave in the current "
+            "environment only the one you want to use for the '{format_name}' format."
+        )
+    else:
+        # conflict with a pandas connector
+        msg = (
+            f"The package `{package_to_load}` registers `{func_name}`, which is "
+            "already provided by pandas. The plugin will be ignored."
+        )
+
+    warnings.warn(msg, UserWarning, stacklevel=1)
+
+
 def load_io_plugins():
     """
     Looks for entrypoints in the `dataframe.io` group and creates the
     corresponding pandas I/O methods.
     """
+    loaded_plugins = {}
+
     for dataframe_io_entry_point in entry_points().get("dataframe.io", []):
         format_name = dataframe_io_entry_point.name
         io_plugin = dataframe_io_entry_point.load()
 
         if hasattr(io_plugin, "reader"):
-            if hasattr(pd, f"read_{format_name}"):
-                raise RuntimeError(
-                    "More than one installed library provides the "
-                    "`read_{format_name}` reader. Please uninstall one of "
-                    "the I/O plugins providing connectors for this format."
+            func_name = f"read_{format_name}"
+            if hasattr(pd, func_name):
+                _warn_conflict(
+                    f"pandas.{func_name}", format_name, loaded_plugins, io_plugin
+                )
+                delattr(pd, func_name)
```

Comment:
Is there any risk we'll remove anything unintentional?

Reply:
I don't think it should happen. I'll think if we can detect the conflicts before registering anything. I think it's tricky the way it's implemented now, but I think it's easier if we use separate entrypoints for readers and writers, which can be a good idea.

If the general concept seems fine, happy to improve the implementation.